ribs/archives/_grid_archive.py

```diff
@@ -155,6 +155,55 @@ def boundaries(self):
         """
         return self._boundaries
 
+    def retessellate(self, new_dims):
+        """Initializes a new archive with ``new_dims`` and re-inserts
+        solutions from the old archive.
+
+        Note that if the new grid resolution is smaller than the old grid
+        resolution, some solutions originally from different cells may end up
+        being assigned to the same cell. Since only a single highest-objective
+        elite is allowed in each cell, some solutions may be dropped.
+
+        Also note that the current implementation does not support archive
+        thresholds. The thresholds within each cell should correspond to how
+        well the measure space within that cell has been explored, and thereby
+        should correspond to the measure space volume within that cell. While
+        this definitely changes after retessellating, it is currently not clear
+        what the new threshold should be.
+
+        Args:
+            new_dims (array-like of int):  Number of cells in each
+            dimension of the measure space, e.g. ``[20, 30, 40]`` indicates
+            there should be 3 dimensions with 20, 30, and 40 cells. The format
+            is similar as the ``dims`` argument in the constructor.
+
+        Returns:
+            GridArchive: A new GridArchive with the new tessellation.
+        """
+        if not np.isclose(self.learning_rate, 1):
+            raise NotImplementedError(
+                "Cannot retessellate an archive with learning rate.")
+
+        new_archive = GridArchive(
```

Comment:
I foresee this method being pretty hard to maintain. Because now whenever we change the definition of GridArchive, we have to come back here and update this method. Not quite sure how to overcome this, though. Thoughts?

Reply:
My concern is that it's a maintenance burden because every time we update the definition of `GridArchive`, we have to remember to modify this definition here.